app/models/case_log.rb

```diff
@@ -219,25 +225,32 @@ def set_derived_fields
     self.hhmemb = other_hhmemb + 1 if other_hhmemb.present?
     self.renttype = RENT_TYPE_MAPPING[rent_type]
     self.lettype = "#{renttype} #{needstype} #{owning_organisation['Org type']}" if renttype.present? && needstype.present? && owning_organisation["Org type"].present?
-    self.is_la_inferred = false if is_la_inferred.nil?
-    if property_postcode.blank? || postcode_known == "No"
-      reset_location_fields
-    else
-      inferred_la = get_inferred_la(property_postcode)
-      self.is_la_inferred = inferred_la.present?
-      self.la = inferred_la if inferred_la.present?
-    end
-    if property_postcode.present?
-      self.postcode = UKPostcode.parse(property_postcode).outcode
-      self.postcod2 = UKPostcode.parse(property_postcode).incode
-    else
-      self.postcode = self.postcod2 = nil
-    end
     self.totchild = get_totchild
     self.totelder = get_totelder
     self.totadult = get_totadult
   end
 
+  def process_postcode_changes!
+    return if property_postcode.blank?
+
+    self.postcode_known = "Yes"
+    inferred_la = get_inferred_la(property_postcode)
+    self.is_la_inferred = inferred_la.present?
+    self.la = inferred_la if inferred_la.present?
```

Comment:
We want to keep the previously set LA if the new postcode doesn't infer a new one.

Reply:
That seems counter-intuitive? This only runs if the `postcode` value has changed and if that's changed surely the previously inferred `la` value is likely to be incorrect?

Reply:
It could be incorrect, but the alternative would be removing an answer which we added previously. When doing some tests with Kat, it is weird when the LA suddenly disappears when you change the postcode. Given we can answer any question, in any order, I don't think we should ever delete something entered. The case where the postcode is valid, but can't be inferred, while the user also previously selected the wrong LA sounds quite far fetched to me. Happy to chat :)